### ISSUE TITLE
Ref safety analysis for extension properties produces different results based on syntax used

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#79654]
**Version Used**: 

**Steps to Reproduce**:

Compile the following code:
```
using System;

ref struct S
{
    public Span<int> field;
}

static class E
{
    extension(S s)
    {
        public S Property { get => throw null!; set => throw null!; }
    }

    extension(ref S s)
    {
        public S RefReceiverProperty { get => throw null!; set => throw null!; }
    }

    public static void Test1(S s)
    {
        var x = new S {  field = stackalloc int[10] };
        s.Property = x; // error
        E.set_Property(s, x); // no errors

        s.RefReceiverProperty = x; // no errors
        E.set_RefReceiverProperty(ref s, x); // error
    }
}
```

**Diagnostic Id**:

`CS8352`, `CS8350`

**Expected Behavior**:
The ref safety analysis doesn't depend on the syntax used to call the `set` accessors.
In both pairs of calls either the ref safety diagnostic is reported in both, or not reported in both.

**Actual Behavior**:
Compilation errors depend on the syntax used to call extension properties' set accessors despite producing equivalent IL code.

**Comment by RikkiGibson** [comment]
I removed the code block around the commit hash so github will format it as a link

It is possible #79447 will fix this, we should add a test.

**Comment by 333fred** [comment]
#79447 will fix the first one, but only incidentally. The second one is more complicated:

https://github.com/dotnet/roslyn/blob/f22f4db8aae84983a41c8b853b4ef4dfc3215c18/src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs#L1609

Nothing tells `GetValEscape` here to use the setter, so it does not. It uses the getter, which has a different escape value.

**Comment by 333fred** [comment]
I expect this is an issue with indexers as well.
